engine/execution/state/state.go

```diff
@@ -400,15 +395,6 @@ func (s *state) saveExecutionResults(
 	ctx context.Context,
 	result *execution.ComputationResult,
 ) (err error) {
-	// make sure only one saving operation is running at a time
-	// this is to prevent the reads within the saving operation to be dirty
```

Comment:
There is no other read except when saving own receipts, which is now protected with a lock, so the lock here is unnecessary, can be removed.